general/device.cpp

```diff
@@ -504,6 +508,18 @@ void Device::Setup(const int device)
          CeedDeviceSetup(device_option);
       }
    }
+   if (Allows(Backend::CEED_HIP))
+   {
+      if (!device_option)
+      {
+         // NOTE: libCEED's /gpu/hip/ref backend is non-deterministic!
+         CeedDeviceSetup("/gpu/hip/ref");
```

Comment:
BTW, my PR in libCEED earlier today was so we can use `/gpu/hip` here and get the fastest (lowest priorityvvalue) HIP backend to date as we make new ones like is node with the CPU impl

Reply:
Yes, I was just following what CUDA had, but since HIP support still has a long way to go in terms of development, it would probably make sense to not have to remember to change this all the time (though we'd still have to update the non-deterministic comment, if it's important to have it).

Reply:
Yeah, it comes into play for the MFEM test suite, I believe, and perhaps some applications

I'm not sure why the GPU impl doesn't do the same, but it probably should

Reply:
This has been changed in 28d366b77c726963ac69d930685b0b664d62fdfb 
I also went ahead and removed the comment about non-deterministic behavior in anticipation of libCEED's PR 623 being merged before this one.